l1-contracts-foundry/script/DeployL1.s.sol

```diff
@@ -0,0 +1,703 @@
+        addresses.stateTransition.stateTransitionProxy = contractAddress;
+    }
+
+    function registerStateTransitionManager() internal {
+        Bridgehub bridgehub = Bridgehub(addresses.bridgehub.bridgehubProxy);
+        vm.broadcast();
+        bridgehub.addStateTransitionManager(addresses.stateTransition.stateTransitionProxy);
+        console.log("StateTransitionManager registered");
+    }
+
+    function setStateTransitionManagerInValidatorTimelock() internal {
+        ValidatorTimelock validatorTimelock = ValidatorTimelock(addresses.validatorTimelock);
+        vm.broadcast();
+        validatorTimelock.setStateTransitionManager(
+            IStateTransitionManager(addresses.stateTransition.stateTransitionProxy)
+        );
+        console.log("StateTransitionManager set in ValidatorTimelock");
+    }
+
+    function deployDiamondProxy() internal {
+        Diamond.FacetCut[] memory facetCuts = new Diamond.FacetCut[](1);
+        facetCuts[0] = Diamond.FacetCut({
+            facet: addresses.stateTransition.adminFacet,
+            action: Diamond.Action.Add,
+            isFreezable: false,
+            selectors: Utils.getAllSelectors(addresses.stateTransition.adminFacet.code)
+        });
+        Diamond.DiamondCutData memory diamondCut = Diamond.DiamondCutData({
+            facetCuts: facetCuts,
+            initAddress: address(0),
+            initCalldata: hex""
+        });
+        bytes memory bytecode = abi.encodePacked(
+            type(DiamondProxy).creationCode,
+            abi.encode(config.chainId, diamondCut)
+        );
+        address contractAddress = deployViaCreate2(bytecode);
+        console.log("DiamondProxy deployed at:", contractAddress);
+        addresses.stateTransition.diamondProxy = contractAddress;
+    }
+
+    function deployErc20BridgeProxy() internal {
+        bytes memory bytecode = abi.encodePacked(
+            type(TransparentUpgradeableProxy).creationCode,
+            abi.encode(addresses.bridgehub.bridgehubProxy, addresses.transparentProxyAdmin, bytes(hex""))
+        );
+        address contractAddress = deployViaCreate2(bytecode);
+        console.log("Erc20BridgeProxy deployed at:", contractAddress);
+        addresses.bridges.erc20BridgeProxy = contractAddress;
+    }
+
+    function deploySharedBridgeContracts() internal {
+        deploySharedBridgeImplementation();
+        deploySharedBridgeProxy();
+        registerSharedBridge();
+    }
+
+    function deploySharedBridgeImplementation() internal {
+        bytes memory bytecode = abi.encodePacked(
+            type(L1SharedBridge).creationCode,
+            // solhint-disable-next-line func-named-parameters
+            abi.encode(
+                config.tokens.tokenWethAddress,
+                addresses.bridgehub.bridgehubProxy,
+                addresses.bridges.erc20BridgeProxy,
+                config.eraChainId,
+                addresses.bridges.erc20BridgeImplementation,
```

Comment:
@kelemeno I am not sure what this address should be. In the original script it is the same as here, but the param itself is:

```
    /// @dev The address of legacy L1 ERC20 bridge.
    address immutable eraErc20BridgeAddress;
```

while for the L1 erc20 bridge we already have this param `legacyBridge`. Considering that the only use case for `eraErc20BridgeAddress` is the following line:

```
        l2BridgeAddress[eraChainId] = eraErc20BridgeAddress;
```

I suggest renaming this variable and fixing both the original script and this one

Reply:
I didn't add this value. We now have two inputs and also two immutables for the legacy/erc20Bridge. We should only have the ERC20BridgeProxy address.

Reply:
This commit introduced this change:
https://github.com/matter-labs/era-contracts/commit/90fa37f70fc3be3556c116b43537c17f057cf45c#diff-c8117da0011136e4b8f35554254e91cfcb71a907b7cec00fac33557c437d2e88

Reply:
Fixed in 3fd43fa0a9c0de7221a80c8a3302f35c862b0daa